HeterogeneousCore/AlpakaInterface/interface/prefixScan.h

```diff
@@ -0,0 +1,235 @@
+#ifndef HeterogeneousCore_AlpakaInterface_interface_prefixScan_h
+#define HeterogeneousCore_AlpakaInterface_interface_prefixScan_h
+// Active version
+
+#include <alpaka/alpaka.hpp>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaInterface/interface/workdivision.h"
+#include "FWCore/Utilities/interface/CMSUnrollLoop.h"
+namespace cms {
+  namespace alpakatools {
+
+    template <typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
+    constexpr bool isPowerOf2(T v) {
+      // returns true iif v has only one bit set.
+      while (v) {
+        if (v & 1)
+          return !(v >> 1);
+        else
+          v >>= 1;
+      }
+      return false;
+    }
```

Comment:
This might be faster, since it is not iterative:
```c++
    // returns true if v is positive and is a power of two
    template <typename T, typename = std::enable_if_t<std::is_integral_v<T>>>
    constexpr bool isPowerOf2(T v) {
      return (v > 0) and (v & v-1) == 0;
    }
```